### ISSUE TITLE
Features to add to GUI Framework

### ISSUE THREAD

**Issue by brechmos-stsci** [spacetelescope/cubeviz#91]
Image Display 
  * Image Viewer (Flux, Error or DQ)
    - [ ] Scale bars on images
    - [ ] Mouse click to change the scale
    - [ ] Edit axis labels (e.g., by clicking them)
    - [ ] Text or region overlays
    - [ ] Cursor shows flux values, and RA/DEC for cursor location
    - [ ] Change image view from RA/DEC to RA/wavelength or DEC/wavelength
    - [ ] Display the wavelength of the current image (#46)
    - [ ] Create checkbox for aligning / syncing of the 3 image panels (one per panel)

  * Cubeviz menubar
    - [ ] Image scaling (manual, log/linear, to 99%)
    - [ ] Color scheme for image display  (e.g., atlas3d or import)
    - [ ] Indictaotr for active window (which one fo flux, error or dq is active)
    - [ ] Collapse image view to certain wavelength range (median, average, total flux)
    - [ ] Button/menu for spatial smoothing ( #74)    
    - [ ] Basic arithmetic operations (#80)
    - [ ] Add text above channel and wavelength selection/viewer to show "Channel" and "Wavelength"
    - [ ] Add text above three image dropdowns and also add text into the plot showing what was selected
  
  * Has been implemented but should be checked
    - [ ] Zoom in and out
    - [ ] Export (spv, eps, pdf)

Spectra Display
  * Spectral Panel Interaction 
    - [ ] Edit axis labels
    - [ ] Two wavelength axes (rest and observed)
    - [ ] Select wavelength range in spectrum display and linemap in image display (linemap would be mean, median or mode)
     - [ ] Display a vertical line in the spectrum showing the wavelength of the current image

  * Spectral Display Menubar
    - [ ] Toggle box for DQ
    - [ ] Redshift text box for rest wavelength axis
    - [ ] Spectrum collapse via mean, median, mode
    - [ ] Select/cut/chop spectrum for further processing (#59)
    - [ ] Button/menu for generic apply function across spaxels ( #78 )
    - [ ] Button/menu for spectral smoothing ( #74)

  * Has been implemented but should be checked
    - [ ] Toggle different spectra on and off
    - [ ] Alpha channel for each spectra
 
Image / Spectra Interaction (Left Hand Panel Level ??? )
  - [ ] Select multiple wavelength ranges in spectra display and allow arithemetic operations on resultant images  (e.g., to remove background/sky)

Advanced Option Panel
  - [ ] 2D Moment maps (#90)

![image](https://user-images.githubusercontent.com/18348847/32556849-865a13ae-c46e-11e7-8abf-fdb99db05236.png)

**** Will add Bethan tagged when they are added to this repo

CC @kassin @hcferguson @drdavella  @drlaw1558

**Comment by brechmos-stsci** [comment]
cc @bethanjames @drlaw1558

**Comment by brechmos-stsci** [comment]
@drdavella and I spent some time today going through cubeviz, in part to see the structure and to make some small changes to the UI.  @astrofrog made a fix or two and now cubeviz layout comes up properly.  @drdavella and I made some changes 1) so the 3 image panel+spectra is the default view and 2) to the slider (added a couple of text boxes for slice # and wavelength value of the current slice -- PR coming on that).  

Tomorrow we are going to go through the list at the top of this issue and determine which things will need to be done before the November sprint in order to have the proper GUI interaction.

cc @kassin @hcferguson @robelgeda
